bin/critterget.py

```diff
@@ -156,19 +156,29 @@ def getCrashSummary(appId, appName):
     mystime = scopetime()
 
     crashattrs = "hash,lastOccurred,sessionCount,uniqueSessionCount,reason,status,displayReason,name"
-    crashdata = apicall("app/%s/crash/summaries" % appId, "lastOccurredStart=%s" % mystime)
-    CrashDict = {}
-    for x,y in enumerate(crashdata):
-        printstring = u'{} MessageType="CrashSummary" appId={} appName="{}" '.format(DATETIME_OF_RUN, appId, appName)
-        slist = crashattrs.split(",")
-        for atname in slist:
-            printstring += u'{}="{}" '.format(atname, crashdata[x][atname])
-            if atname == "hash" : CrashDict[crashdata[x][atname]]= appName
-        print printstring
 
+    http_code, crashdata = apicall_with_response_code("app/%s/crash/summaries" % appId, "lastOccurredStart=%s" % mystime)
+    if http_code == 500:
+        retry = 1
+        while http_code == 500 and retry < MAX_RETRY:
+            http_code, crashdata = apicall_with_response_code("app/%s/crash/summaries" % appId, "lastOccurredStart=%s" % mystime)
```

Comment:
This is logically correct, but the `if` has some duplicate logic.  It should be unnecessary if you juggle things around a little.  One way would be:

```
http_code = 500
retry = 1
while http_code == 500 and retry < MAX_RETRY:
    http_code, crashdata = ...
```

Another way would just have a `while http_code` and do a `break` if `retry` exceeds your `MAX_RETRY`, since python doesn't have a do...while.  But whatever.  See if you can do it without that `if` statement that duplicates the apicall.